src/v/pandaproxy/server.cc

```diff
@@ -96,13 +96,40 @@ struct handler_adaptor : ss::httpd::handler_base {
                        [this, rq{std::move(rq)}, rp{std::move(rp)}]() mutable {
                            if (_ctx.as.abort_requested()) {
                                set_reply_unavailable(*rp.rep);
+                               _probe.increment_5xx();
                                return ss::make_ready_future<
                                  std::unique_ptr<ss::reply>>(std::move(rp.rep));
                            }
                            return _handler(std::move(rq), std::move(rp))
                              .then([](server::reply_t rp) {
                                  set_mime_type(*rp.rep, rp.mime_type);
                                  return std::move(rp.rep);
+                             })
+                             .handle_exception_type(
+                               [this](pandaproxy::parse::exception_base& ex) {
```

Comment:
I feel like there must be more than parse errors that can result in an error.

What about `pandaproxy::json::exception_base`, for example?

Does this intercept the path before `exception_replier`?

Reply:
> I feel like there must be more than parse errors that can result in an error.
> 
> What about `pandaproxy::json::exception_base`, for example?

I did not test those out. Will do today. Actually, do you have recommendation on how to organically get an error of that type? Otherwise I'll just hard-code in a `throw`


> Does this intercept the path before `exception_replier`?

From my experiments yesterday it seemed that all REST failures eventually go through [the handler](https://github.com/redpanda-data/redpanda/blob/f8d29fbd87ee0583ca63328648902f9d78b1ca4c/src/v/pandaproxy/server.cc#L102) so there may not be a need.

Reply:
> > Does this intercept the path before `exception_replier`?
> 
> From my experiments yesterday it seemed that all REST failures eventually go through [the handler](https://github.com/redpanda-data/redpanda/blob/f8d29fbd87ee0583ca63328648902f9d78b1ca4c/src/v/pandaproxy/server.cc#L102) so there may not be a need.

This should be the case because the `exception_replier` is registered to each route. Likewise a `handler_adaptor` is attached to each route.
```
    auto* handler = new handler_adaptor(
      _pending_reqs,
      _ctx,
      std::move(r.handler),
      r.path_desc,
      _public_metrics_group_name);
    r.path_desc.set(_server._routes, handler);
```
So we could capture errors either via exception_replier or handler_adaptor. Conceptually, I don't see a difference between either approaches but perhaps using the handler_adaptor is more "hacky" ?